### PR TITLE
enable strict deabstraction on crashers.swift (SR-8303)

### PR DESCRIPTION
**Pull request by marcrasi**
This fixes two things that were breaking strict deabstraction in crashers.swift:
1. Strict deabstraction makes some errors happen earlier, which was blocking partitioning from happening in crashers.swift. I moved the tests with those errors to a new file.
2. formGraphOp does not have a check for unsupported attribute types, so the unsupported types get into graph lowering and cause an assertion failure. I added a check for unsupported attribute types in formGraphOp.

### CODE REVIEW

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFDeabstraction.cpp [hunk 1]
```diff
@@ -2043,7 +2043,27 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
     switch (operandClass.second) {
     case SILTensorOpInfo::OperandClass::Input:
     case SILTensorOpInfo::OperandClass::InputElt:
+      assert(0 && "Input classes cannot exist for attributes");
     case SILTensorOpInfo::OperandClass::Normal:  // No modifier.
+      switch (constValue.getKind()) {
```
Comment (bgogul) [inline lib/SILOptimizer/Mandatory/TFDeabstraction.cpp]: Can you refactor this into a function or lambda?

Comment (marcrasi) [reply lib/SILOptimizer/Mandatory/TFDeabstraction.cpp]: done

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/TFDeabstraction.cpp [hunk 2]
```diff
@@ -2014,6 +2014,34 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
...
     }
 
+    // Emits a diagnostic and returns true if the value is invalid for a shape
+    // attr.
+    auto verifyNormalAttr = [&](SymbolicValue constValue) -> bool {
+      switch (constValue.getKind()) {
+      case SymbolicValue::Unknown:
+      case SymbolicValue::UninitMemory:
+      case SymbolicValue::Address:
+        assert(0 && "Cannot happen");
```
Comment (bgogul) [inline lib/SILOptimizer/Mandatory/TFDeabstraction.cpp]: More meaningful message?

**Review by bgogul — APPROVED** [review]
Nice! Just one nit.

File: lib/SILOptimizer/Mandatory/TFDeabstraction.cpp (hunk 2 above)
Comment (marcrasi) [reply lib/SILOptimizer/Mandatory/TFDeabstraction.cpp]: done!

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Comment by marcrasi** [comment]
@swift-ci please test tensorflow

**Pull request merged**